Common/Securities/NonShortableCashAccountMarginModel.cs

```diff
@@ -0,0 +1,158 @@
+        /// </summary>
+        /// <param name="cashBook">The portfolio's cashbook</param>
+        public NonShortableCashAccountMarginModel(CashBook cashBook)
+        {
+            _cashBook = cashBook;
+        }
+
+        /// <summary>
+        /// Cash accounts require the full value of the order plus any fees
+        /// </summary>
+        /// <param name="security"></param>
+        /// <param name="order">The order to be executed</param>
+        /// <returns>The total margin in terms of the currency quoted in the order</returns>
+        public decimal GetInitialMarginRequiredForOrder(Security security, Order order)
+        {
+            var fees = security.FeeModel.GetOrderFee(security, order);
+            var marginRequiredInAccountCurrency = order.GetValue(security) + fees;
+
+            switch (order.Direction)
+            {
+                case OrderDirection.Hold:
+                case OrderDirection.Buy:
+                    // convert the order value to the quote currency
+                    return marginRequiredInAccountCurrency / security.QuoteCurrency.ConversionRate;
+
+                case OrderDirection.Sell:
+                    // remaining margin in units of the base currency ( can't sell what we don't have )
+                    var baseCurrency = security as IBaseCurrencySymbol;
+                    if (baseCurrency != null)
+                    {
+                        // we have this much 'cash' that can be sold
+                        var feesInBaseCurrency = _cashBook.Convert(fees, CashBook.AccountCurrency, baseCurrency.BaseCurrencySymbol);
+                        return order.AbsoluteQuantity + feesInBaseCurrency;
+                    }
+
+                    // assuming these are 'units' of stock priced in the account currency
+                    return marginRequiredInAccountCurrency;
+
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        /// <summary>
+        /// Gets the margin cash available for a trade in units of the currency required.
+        /// When increasing a position, this is the quote currency, when decreasing a position this is the base currency
+        /// </summary>
+        /// <param name="portfolio">The algorithm's portfolio</param>
+        /// <param name="security">The security to be traded</param>
+        /// <param name="direction">The direction of the trade</param>
+        /// <returns>The margin available for the trade</returns>
+        public decimal GetMarginRemaining(SecurityPortfolioManager portfolio, Security security, OrderDirection direction)
+        {
+            switch (direction)
+            {
+                case OrderDirection.Hold:
+                case OrderDirection.Buy:
+                    // increasing position, purchasing in units of the quote currency
+                    return security.QuoteCurrency.Amount;
```

Comment:
I think the only thing missing from this `GetMarginRemaining` implementation is checking open orders. This is kind of tricky, since we need to check for orders not only for `security` but also for all other securities that have the currency while keeping in mind the side of the trade.